### DISCUSSION TITLE
Can I use the lavamoat webpack plugin for a Typescript + React + Nextjs project?

### DISCUSSION THREAD

**Discussion by schmanu** [LavaMoat/LavaMoat#832] [Q&A]
I am working on the [Safe Wallet Web app](https://github.com/safe-global/safe-wallet-web) and would like to experiment with integrating the Lavamoat Webpack plugin.

We use React, Typescript and NextJS with Webpack. I am currently stuck trying to create the policy file for our app.
I am running `npx lavamoat --autopolicy` but wasn't able to pass a proper entry point to it. I tried using the built chunks and the uncompiled .tsx files but neither worked.
Is the lavamoat plugin already ready to be combined with out stack? Do you have an example for a nextjs react project?

Any help is highly appreciated.

**Comment by compojoom** [comment]
@naugtur have you guys tested the webpack plugin with nextjs? I see that you were active in some nextjs issues. 

We tried today to add the pluign to the nextjs config, but the build was failing with 

> LavaMoatPlugin: Failed to inspect module next/dist/compiled/next-server/pages.runtime.prod.js for policy generation:
>  Cannot read properties of undefined (reading 'length')
>  If the file is not intended to be valid JavaScript, consider excluding it using LavaMoat.exclude loader.
> 
> LavaMoatPlugin: Progress reported 'finish' but the next step was expected to be 'pathsCollected'
> 
> LavaMoatPlugin: Progress reported 'finish' but the next step was expected to be 'generatorCalled' 

I tried modifying the whole module.rules array and add a Lavamoat.exclude, but the rule that is adding the next-server apparently has a loader and there I can't use "use".

Any hints?

**Reply by naugtur** [reply]
next is very complex and unlikely to work out of the box. If you're willing to collaborate I can guide you through the process of finding incompatibilities. I myself don't have much experience with next.js build system so pairing with someone who works with it is my only way forward. 

I'm off for a bit but should be more available in a week or so

**Reply by compojoom** [reply]
Hey @naugtur, good idea. Let's collaborate on this! I've sent you a mail.
